esp/services/ws_workunits/ws_workunitsQuerySets.cpp

```diff
@@ -2876,6 +2934,59 @@ class QueryCloner
     {
         queryDirectory.set(dir);
     }
+
+    void addToBePublished(const char *wuid, const char *name, bool makeActive, const char *userid, IPropertyTree *query)
+    {
+        OwnedPTree entry = toBePublished->addPropTree("Publish");
+        entry->setProp("@wuid", wuid);
+        entry->setProp("@name", name);
+        entry->setPropBool("@makeActive", makeActive);
+        entry->setProp("@userid", userid);
+        entry->addPropTree("Info", createPTreeFromIPT(query));
+
+        Owned<IAttributeIterator> aiter = query->getAttributes();
+        OwnedPTree attrs = entry->addPropTree("Attrs");
```

Comment:
doesn't look like correct to own this (or the one on line 2940 above).

addPropTree returns a unlinked ptr to new IPT added, owning here is afaics, going to destroy it, whilst it remains in the parent.